packages/react/src/components/tag/tag.tsx

```diff
@@ -56,202 +125,253 @@ function getIconSize(isMobile: boolean): string {
         }
     }
 
     ${focus};
 `;
 
-const Container = styled.span<ContainerProps>`
-    align-items: center;
-    background-color: ${({ theme }) => theme.component['tag-background-color']};
+function getClickableStyle(
+    { $clickable, $selected, $tagSize }: TagContainerProps,
+): FlattenInterpolation<ThemeProps<ResolvedTheme>> | false {
+    return $clickable && css`
+        font-family: var(--font-family);
+        padding-left: ${isSmall($tagSize) ? 'var(--spacing-1x)' : 'var(--spacing-1halfx)'};
+        padding-right: ${isSmall($tagSize) ? 'var(--spacing-1x)' : 'var(--spacing-1halfx)'};
+
+        &:hover {
+            background-color: ${({ theme }) => ($selected ? theme.component['tag-default-selected-hover-background-color'] : theme.component['tag-default-hover-background-color'])};
+            border-color: ${({ theme }) => ($selected ? theme.component['tag-default-selected-hover-border-color'] : theme.component['tag-default-hover-border-color'])};
+        }
+
+        ${focus};
+    `;
+}
 
-    /* TODO fix with next thematization gray50 */
+const TagContainer = styled.div<TagContainerProps>`
+    align-items: center;
+    background-color: ${(props) => getTagColors(props, 'background-color')};
+    border: 1px solid ${(props) => getTagColors(props, 'border-color')};
     border-radius: ${getBorderRadius};
-    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.component['tag-box-shadow-color']};
     display: inline-flex;
+    justify-content: center;
     padding: ${getPadding};
 
     & + & {
         margin-left: var(--spacing-1x);
     }
 
-    ${DeleteButton} {
-        margin-right: calc(-1 * var(--spacing-half));
-    }
+    ${({ $selected, theme }) => $selected && css`
+        background-color: ${theme.component['tag-default-selected-background-color']};
+        border: 1px solid ${theme.component['tag-default-selected-border-color']};
+    `}
 
     ${getClickableStyle};
-
-    ${focus};
 `;
 
 export const Tag = forwardRef(({
     className,
     iconName,
-    onClick,
-    onDelete,
     size = 'medium',
+    color = 'default',
+    selected = false,
     value,
+    onClick,
+    onDelete,
 }: TagProps, ref: Ref<HTMLElement>) => {
-    if (onClick && onDelete) {
-        throw new Error('Only one of onClick or onDelete can be provided.');
-    }
-
     const { t } = useTranslation('tag');
     const { isMobile } = useDeviceContext();
+
+    const isDeletable = !!onDelete;
+    const isClickable = !!onClick;
+    const hasIcon = !!iconName;
+    const currentColor = (isDeletable || isClickable || hasIcon) ? 'default' : (color ?? 'default');
+    const [isSelected, setSelected] = useState<boolean>(isDefault(currentColor) && selected);
     const hasIconLabel = !(value.label.toLowerCase() === iconName?.toLowerCase());
+    const shortenedLabel = value.label.length > 20 ? `${value.label.slice(0, 17)}…` : value.label;
```

Comment:
Est-ce qu'on pourrait utiliser `text-overflow: ellipsis` pour ça?

Reply:
Ca faut 10 fois je relis Notion et finalement je crois faut pas mettre de text-ellipsis du tout. La regle du 20 characteres est une recommendation (should), mais ca peut etre plus long si on se base sur le spec suivant.

![image](https://github.com/kronostechnologies/design-elements/assets/3039571/92fa4a6f-87a5-48e3-8471-7c59afbf518f)

Reply:
Je laisserais ca au App de faire eux-meme le truncate et ellipsis.